### PR TITLE
Logging: Drop remaining Settings log ctor

### PR DESCRIPTION
**Pull request by nik9000**
Drops the last logging constructor that takes `Settings` because it is
no longer needed.

Watcher goes through a lot of effort to pass `Settings` to `Logger`
constructors and dropping `Settings` from all of those calls allowed us
to remove quite a bit of log-based ceremony from watcher.

### CODE REVIEW

File: x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/WatcherServiceTests.java [hunk 1]
```diff
@@ -231,7 +231,7 @@ public void testPausingWatcherServiceAlsoPausesTriggerService() {
         Watch watch = mock(Watch.class);
         when(watch.trigger()).thenReturn(trigger);
         when(watch.condition()).thenReturn(InternalAlwaysCondition.INSTANCE);
-        ExecutableNoneInput noneInput = new ExecutableNoneInput(logger);
+    ExecutableNoneInput noneInput = new ExecutableNoneInput();
```
Comment (cbuescher) [inline x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/WatcherServiceTests.java]: nit: indentation

File: x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/test/TimeWarpedWatcher.java [hunk 2]
```diff
@@ -36,8 +35,7 @@
 
     public TimeWarpedWatcher(final Settings settings, final Path configPath) throws Exception {
         super(settings, configPath);
-        Logger logger = Loggers.getLogger(TimeWarpedWatcher.class, settings);
-        logger.info("using time warped watchers plugin");
+        LogManager.getLogger(TimeWarpedWatcher.class).info("using time warped watchers plugin");
```
Comment (cbuescher) [inline x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/test/TimeWarpedWatcher.java]: I don't know how often this class is created, but maybe it makes sense to store a reference to the logger for later use if this class is instantiated more often.

**Review by cbuescher — APPROVED** [review]
I left two nits but nothing big, rest LGTM

**Comment by nik9000** [comment]
Thanks for reviewing @cbuescher!

**Pull request merged**